src/main/scala/viper/gobra/translator/transformers/hyper/SIFExtendedTransformer.scala

```diff
@@ -0,0 +1,2165 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2023 ETH Zurich.
+
+package viper.gobra.translator.transformers.hyper
+
+import viper.gobra.backend.BackendVerifier
+import viper.gobra.translator.transformers.ViperTransformer
+import viper.silver.ast._
+import viper.silver.ast.utility.Simplifier
+import viper.silver.verifier.{AbstractError, errors}
+import viper.silver.verifier.errors.{AssertFailed, ErrorNode}
+import viper.gobra.translator.util.{ViperUtil => vu}
+
+import scala.collection.immutable.HashSet
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+trait SIFExtendedTransformer {
+  object Config {
+    /** If true, don't generate all the control flow variables, just the ones needed for each method. */
+    var optimizeControlFlow: Boolean = true
+    /** If true, try to bunch as many statements into a single if statement which was introduced for checking active
+      * executions, instead of having one if stmt per original statement. */
+    var optimizeSequential: Boolean = true
+    /** If true, add an 'assume p1' at the beginning of each method, to cut down on redundant paths the
+      * verification could consider */
+    var optimizeRestrictActVars: Boolean = true
+    /** Applications of the functions which have an entry here, will be replaced by the expression
+      * determined by the entry in the second execution. */
+    var primedFuncAppReplacements: mutable.HashMap[String, (FuncApp, Exp, Exp) => Exp] = new mutable.HashMap
+    /** Set this to only transform methods that contain relational assertions somewhere in their spec or body.
+      * May lead to invalid programs when such a method calls another methods that does contain such specs.
+      */
+    var onlyTransformMethodsWithRelationalSpecs: Boolean = false
```

Comment:
@ArquintL I guess setting this to true should allow us to keep the default for SIF. I will change it once I start working on this pR

Reply:
That's not sufficient. Instead of checking which methods have relational spec and which do not, a first attempt would be to decide this for the entire input program. To keep this on a per-method level, we would first need to address the problem mentioned in the comment above namely that a non-SIF method calling a SIF method (or vice-versa) fails as the number of parameters / arguments does not match.
A similar issue occurs if we would mix methods encoded with optimized SIF encoding (enforcing low control flow) and the more complete SIF encoding as the latter one uses additional parameters (namely the activation variables) and also uses a different ordering for the parameters belonging to the 2nd execution (which is just an implementation detail)